arcgis-ios-sdk-samples/Edit data/Create mobile geodatabase/CreateMobileGeodatabaseViewController.swift

```diff
@@ -0,0 +1,203 @@
+                        let featureLayer = AGSFeatureLayer(featureTable: table)
+                        // Add the feature layer to the map's operational layers.
+                        self.mapView.map?.operationalLayers.add(featureLayer)
+                        self.featureCountLabel.text = "Number of features added: 0"
+                    }
+                } else if let error = error {
+                    self.presentAlert(error: error)
+                }
+            }
+        }
+    }
+    
+    /// Add a feature at the provided map point.
+    func addFeature(at mapPoint: AGSPoint) {
+        // A dictionary to store the custom attribute.
+        var attributes = [String: Date]()
+        // Create an attribute with the current date.
+        attributes["collection_timestamp"] = Date()
+        // Create a feature with the created attribute and geometry.
+        guard let feature = featureTable?.createFeature(attributes: attributes, geometry: mapPoint) else { return }
+        // Add the feature to the feature table.
+            featureTable?.add(feature) { [weak self] error in
+                guard let self = self, let numberOfFeatures = self.featureTable?.numberOfFeatures else { return }
+                let featureCount = String(numberOfFeatures)
+                // Update the label's text to display the current number of features.
+                self.featureCountLabel.text = String(format: "Number of features added: %@", featureCount)
+                // Enable the view table bar button item.
+                self.viewTableBarButtonItem.isEnabled = true
+                if let error = error {
+                    self.presentAlert(error: error)
+                }
+            }
+    }
+    
+    /// Remove existing operational layers and close the geodatabase.
+    func resetMap() {
+        if mapView.map?.loadStatus == .loaded, let geodatabase = geodatabase {
+            // Close the geodatabase to cease all adjustments.
+            geodatabase.close()
+            // Remove the current feature layers.
+            mapView.map?.operationalLayers.removeAllObjects()
+            // Reset the button's state.
+            viewTableBarButtonItem.isEnabled = false
+            // Create a new mobile geodatabase.
+            createGeodatabase()
+        } else if let error = mapView.map?.loadError {
+            presentAlert(error: error)
+        }
+    }
+    
+    deinit {
+        try? FileManager.default.removeItem(at: temporaryDirectory)
+    }
+    
+    // MARK: UIViewController
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let navigationController = segue.destination as? UINavigationController,
+           let controller = navigationController.topViewController as? MobileGeodatabaseTableViewController {
+            // Query features to list the existing features.
+            featureTable?.queryFeatures(with: AGSQueryParameters()) { [weak self] results, error in
```

Comment:
An async operation shouldn't happen during configuring the UI. It would result in data inconsistency, see the screenshot below - features are added but not shown in the table view.

<img width="444" alt="image" src="https://user-images.githubusercontent.com/9660181/179589297-a55fc91f-a366-400a-ab75-3d87fbfb6ed9.png">

Instead, a few other ideas:

- query the feature table first, and then show table view. When the feature table is querying, maybe use a HUD to block user interaction. Instead of letting the button to trigger the segue, you may manually construct and present the tableVC in the query completion closure
- whenever a feature is added, query the feature table and store the results. so when the segue action is triggered, data is ready to use

Reply:
I added a query features function and manually constructed the tableVC but it removed the navigation bar. Please let me know what you think.